java/ql/automodel/publish.sh

```diff
@@ -49,8 +55,14 @@ fi
 CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 CURRENT_SHA=$(git rev-parse HEAD)
 
-if [ -z "${1:-}" ]; then
-  # If the first argument is empty, use the latest release of codeml-automodel
+if [ "$1" = "--override-release" ] || [ "$2" = "--override-release" ]; then
+  # Check that the current HEAD is downstream from PREVIOUS_RELEASE_SHA
+  if ! git merge-base --is-ancestor "$PREVIOUS_RELEASE_SHA" "$CURRENT_SHA"; then
```

Comment:
what if the last change to `./src/qlpack.yml` was a change that's unrelated to a release being done? would that lead to problems here?

Reply:
I'm unsure. Perhaps in unlikely edge cases. The intent of this check is to prevent the accidental publication of an query-pack that's behind the current one (which could happen if, e.g., someone publishes a "patch" release with `--override-release` and the official `codeql-automodel` release hasn't yet caught up with it).